pkg/storage/minio/minio_connecter.go

```diff
@@ -0,0 +1,80 @@
+package minio
+
+import (
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"go.uber.org/zap"
+)
+
+type Connecter struct {
+	endpoint        string
+	accessKeyID     string
+	secretAccessKey string
+	ssl             bool
+	region          string
+	token           string
+	bucket          string
+	log             *zap.SugaredLogger
+	minioClient     *minio.Client
+}
+
+// NewConnecter creates new MinioSubscriber which will send data to local MinIO bucket
+func NewConnecter(endpoint, accessKeyID, secretAccessKey, region, token, bucket string, ssl bool, log *zap.SugaredLogger) *Connecter {
+	c := &Connecter{
+		region:          region,
+		accessKeyID:     accessKeyID,
+		secretAccessKey: secretAccessKey,
+		token:           token,
+		ssl:             ssl,
+		bucket:          bucket,
+		endpoint:        endpoint,
+		log:             log,
+	}
+
+	return c
+}
+
+// GetClient returns minio client
+func (c *Connecter) GetClient() (*minio.Client, error) {
+	if c.minioClient == nil {
+		var err error
+		c.minioClient, err = c.Connect()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return c.minioClient, nil
+}
+
+// Connect connects to MinIO server
+func (c *Connecter) Connect() (*minio.Client, error) {
```

Comment:
Why we're duplicating code from minio.go file here?

Reply:
actually I got it here to not duplicate it and to be able to use it in the log minio consumer